File: leet/0082-remove-duplicates-from-sorted-list-II/remove-duplicates-from-sorted-list-II.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
    if(head == nil || head.Next == nil) { return head }
    
    dummy := ListNode{Next: head}
    last := &dummy
    current := head
    
    for current != nil {
        if(current.Next != nil && current.Val == current.Next.Val){
            c1 := current
            for c1 != nil && c1.Val == current.Val {
                c1 = c1.Next
            }
            
            last.Next = c1
            current = c1
        } else {
            last = current
            current = current.Next
        }
    }
    
    return dummy.Next
}

File: leet/0082-remove-duplicates-from-sorted-list-II/remove-duplicates-from-sorted-list-II.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    let dummy = new ListNode(-1)
    dummy.next = head
    let pst = dummy

    while(pst.next){
        let next = pst.next
        while(next.next && next.val == next.next.val) next = next.next
        if(next != pst.next) pst.next = next.next
        else pst = pst.next
    }
    
    return dummy.next
};
